### PR TITLE
(BKR-384) & (BKR-385) support for dev puppet-agent installation on osx

### PR DESCRIPTION
**Pull request by anodelman**
(BKR-384) install_puppet_agent_dev_repo needs to support osx
    - able to install dev puppet-agent on osx
    
    example:
    
    install_puppet_agent_dev_repo_on(host,
       { :puppet_agent_sha => 'c2946821d1bcd7e43a34bd1f9cd2f86160ca4fe0',
         :puppet_agent_version => '1.2.1.23.gc294682'    })

(BKR-385) beaker needs to be able to convert mac codename to version number
    
    - add support to the platform object to handle yosemite -> 10.10 and
      mavericks -> 10.9 conversions for osx variant

### CODE REVIEW

File: lib/beaker/dsl/install_utils/foss_utils.rb [hunk 1]
```diff
@@ -1020,6 +1020,10 @@ def install_puppet_agent_dev_repo_on( hosts, opts )
               # - we do not have install_32 set globally
               arch_suffix = should_install_64bit ? '64' : '86'
               release_file = "puppet-agent-x#{arch_suffix}.msi"
+            when /^osx$/
+              onhost_copy_base = File.join('/var', 'root')
+              release_path << "/apple/#{opts[:puppet_collection]}"
```
Comment (doug-rosser) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: My ruby-fu is weaksauce. I am assuming "<<" is not a leftwise shift operator, so what it do?

**Comment by sschneid** [comment]
:+1:

**Comment by doug-rosser** [comment]
:+1: Pending test results

File: lib/beaker/dsl/install_utils/foss_utils.rb [hunk 2]
```diff
@@ -1038,6 +1042,8 @@ def install_puppet_agent_dev_repo_on( hosts, opts )
               result = on host, "echo #{onhost_copied_file}"
               onhost_copied_file = result.raw_output.chomp
               on host, Command.new("start /w #{onhost_copied_file}", [], { :cmdexe => true })
+            when /^osx$/
+              host.install_package("puppet-agent-*")
```
Comment (joshcooper) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: this seems a little risky, if you have more than one dmg on the coordinator (running acceptance on-demand for multiple SHAs), then which dmg will this install?

**Comment by joshcooper** [comment]
Aside from my comment, this PR worked for me. I updated puppet's setup step to do:

```
  when /osx/
    opts = {
      :puppet_collection => 'PC1',
      :puppet_agent_sha => ENV['SHA'],
      :puppet_agent_version => ENV['SUITE_VERSION'] || ENV['SHA']
    }
    install_puppet_agent_dev_repo_on(agent, opts)
  end
```

```
$ be rake ci:test:aio SUITE_VERSION=1.1.1.250.gb9e1783 SHA=b9e178350dd52c75ab98ffd05e9bc739d7609155 CONFIG=config/nodes/osx-1010-x86_64.yaml TESTS=tests/ticket_7101_template_compile.rb
...
Fetching: http://builds.delivery.puppetlabs.net/puppet-agent/b9e178350dd52c75ab98ffd05e9bc739d7609155/repos//apple/PC1/puppet-agent-1.1.1.250.gb9e1783-osx-10.10-x86_64.dmg
  and saving to tmp/repo_configs/osx/puppet-agent-1.1.1.250.gb9e1783-osx-10.10-x86_64.dmg
localhost $ scp tmp/repo_configs/osx/puppet-agent-1.1.1.250.gb9e1783-osx-10.10-x86_64.dmg agent:/var/root {:ignore => }

esm5vdqt28849mt.delivery.puppetlabs.net (agent) 16:26:01$ hdiutil attach puppet-agent-*.dmg
Checksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified   CRC32 $667C33CF
Checksumming GPT Header (Primary GPT Header : 1)…
 GPT Header (Primary GPT Header : 1): verified   CRC32 $1267A7C9
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified   CRC32 $A77BD203
Checksumming  (Apple_Free : 3)…
                    (Apple_Free : 3): verified   CRC32 $00000000
Checksumming disk image (Apple_HFS : 4)…
          disk image (Apple_HFS : 4): verified   CRC32 $1E3B2375
Checksumming  (Apple_Free : 5)…
                    (Apple_Free : 5): verified   CRC32 $00000000
Checksumming GPT Partition Data (Backup GPT Table : 6)…
GPT Partition Data (Backup GPT Table: verified   CRC32 $A77BD203
Checksumming GPT Header (Backup GPT Header : 7)…
  GPT Header (Backup GPT Header : 7): verified   CRC32 $C2130BC7
verified   CRC32 $D2FE4FA3
/dev/disk1          	GUID_partition_scheme
/dev/disk1s1        	Apple_HFS                      	/Volumes/puppet-agent-1.1.1.250.gb9e1783

esm5vdqt28849mt.delivery.puppetlabs.net (agent) executed in 3.13 seconds

esm5vdqt28849mt.delivery.puppetlabs.net (agent) 16:26:04$ installer -pkg /Volumes/puppet-agent-*/puppet-agent-*.pkg -target /
installer: Package name is puppet-agent
installer: Installing at base path /
installer: The install was successful.
...
Beaker completed successfully, thanks.
```

File: lib/beaker/dsl/install_utils/foss_utils.rb (hunk 1 above)
Comment (anodelman) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: String append.

Comment (waynr) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: @doug-rosser http://ruby-doc.org/core-2.2.2/String.html#method-i-3C-3C

File: lib/beaker/dsl/install_utils/foss_utils.rb (hunk 2 above)
Comment (waynr) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: Looks like `onhost_copied_file` might be a suitable alternative.

**Comment by anodelman** [comment]
Updated to be somewhat better at dmg finding, now installs puppet-agent-#{version}*, so it shouldn't attempt to install something unexpected.

**Pull request merged**